File: Net/Chat/MessageProducer.cs
using Confluent.Kafka;
using Confluent.Kafka.Admin;

namespace Chat
{
  internal class MessageProducer : IDisposable
  {
    private readonly string name;
    private readonly IProducer<string, string> producer;

    public MessageProducer(string name)
    {
      this.name = name;
      var producerConfig = new ProducerConfig {
        BootstrapServers = Common.Constants.KafkaHost,
        AllowAutoCreateTopics = true
      };
      producer = new ProducerBuilder<string, string>(producerConfig).Build();
    }

    public void CreateTopicIfNotExists()
    {
      var adminClientBuilder = new AdminClientBuilder(
        new AdminClientConfig {
          BootstrapServers = Common.Constants.KafkaHost
        });

      using var adminClient = adminClientBuilder.Build();
      var topicNames = adminClient.
        GetMetadata(TimeSpan.FromSeconds(2)).Topics.
        Select(x => x.Topic).ToList();
      if (topicNames.Any(n => n == Common.Constants.ChatTopic)) 
        adminClient.CreateTopicsAsync(new TopicSpecification[] {
            new TopicSpecification {
              Name = Common.Constants.ChatTopic 
            } 
        });
    }

    public void SendMessage(string message)
    {
      producer.Produce(
        Common.Constants.ChatTopic,
        new Message<string, string> {
          Key = name,
          Value = message
        });
      producer.Flush();
    }

    public void Dispose()
    {
      producer.Dispose();
    }

  }
}

File: Java/src/main/java/com/example/demo/MessageProducer.java
package com.example.demo;

import java.io.Closeable;
import java.io.IOException;
import java.util.Properties;

import org.apache.kafka.clients.producer.KafkaProducer;
import org.apache.kafka.clients.producer.ProducerConfig;
import org.apache.kafka.clients.producer.ProducerRecord;

public class MessageProducer implements Closeable {

    private final String name;

    private final KafkaProducer<String, String> producer;

    public MessageProducer(String name) {
        this.name = name;
        Properties properties = new Properties();
        properties.put(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, Common.KAFKA_HOST);
        this.producer = new KafkaProducer<>(properties);
    }

    public void sendMessage(String message) {
        ProducerRecord<String,String> record = new ProducerRecord<String,String>(Common.CHAT_TOPIC, this.name, message);
        this.producer.send(record);
    }

    @Override
    public void close() throws IOException {
        this.producer.close();
    }
    
}
